src/plasma/plasma_manager.cc

```diff
@@ -344,9 +337,9 @@ void object_table_subscribe_callback(ObjectID object_id,
                                      const char *manager_vector[],
                                      void *context);
 
-ObjectWaitRequests **object_wait_requests_table_ptr_from_type(
-    PlasmaManagerState *manager_state,
-    int type) {
+std::unordered_map<ObjectID, std::vector<WaitRequest *>, UniqueIDHasher>
```

Comment:
I think you generally don't want to return pointers to STL containers, you should just return the container itself.

I *think* that in this case we want to return a 

```c++
std::unordered_map<ObjectID, std::vector<WaitRequest *>, UniqueIDHasher> &
```

but it'd be good if @pcmoritz could comment as well.

When we use it, it should also be a reference. If you do

```c++
auto x = object_wait_requests_ptr_from_type(manager_state, type);
```

then will `x` be a reference or not?

Reply:
I think you're right, that should work. I'll implement that fix.

Reply:
Yeah, a reference here is better; on the caller side I have a feeling that
```
auto& x = object_wait_requests_ptr_from_type(manager_state, type);
```
is the way to go, but I'm not sure if that works.